Automattic/jetpack
# VideoPress: Fix VideoPress Attachment Deletions on wp.com

Fixes issue where deleting a VideoPress video would not actually delete the VideoPress entry on wp.com. Uses the `pre_delete_attachment` filter to attempt deletion of VideoPress video on wp.com.

There's a big change here in that the attachment will not get deleted unless wp.com successfully deleted the VideoPress video (unless the API returns a `404` which means it was already deleted or not found). This is to prevent any dangling data. One thing I'd like advice on here is how we can tell the user there was an error so they can take appropriate action. At a minimum it'd be great if we could add to the activity log so that happiness can help troubleshoot.

#### Changes proposed in this Pull Request:
* Updates VideoPress attachment deletions to use the wp.com api to actually delete the video from our servers.

#### Jetpack product discussion
pxWta-Zz-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Select a video in the media library and copy it's `guid`.
* Delete the video.
* Check that the video is not found at https://videopress.com/v/{guid}
* Repeat the above steps for videos in the Calypso media library.

**Also test that attachment poster image is deleted:**
* Right click on a video thumbnail and copy the url for the poster image:
<img width="741" alt="Screen Shot 2021-08-31 at 2 19 39 PM" src="https://user-images.githubusercontent.com/789137/131570542-22505569-a8f1-4f28-90ce-c061148c4460.png">
* Delete the video
* Try to access the thumbnail url, it should also be deleted.
